pkg/util/ranger/detacher.go

```diff
@@ -273,8 +273,37 @@ func extractBestCNFItemRanges(sctx *rangerctx.RangerContext, conds []expression.
 			continue
 		}
 		curRes := getCNFItemRangeResult(sctx, res, i)
-		if bestRes == nil || compareCNFItemRangeResult(curRes, bestRes) {
+		if bestRes == nil {
 			bestRes = curRes
+		} else {
+			if curRes != nil && bestRes.rangeResult != nil {
+				if fixcontrol.GetBoolWithDefault(sctx.OptimizerFixControl, fixcontrol.Fix54166, false) {
```

Comment:
Should we enable it be default and don't need a control variable ?
It seems that intersection range is actually an always good way ?

Reply:
You are right that the intersection is always better. The control is to minimize the risk initially for the high complexity of the intersection. We should take it out in few sprints and leave the solution as is or add a heuristic per the design for the complex cases (too many ranges to intersect).

Reply:
I would rather keep it off initially and turn it on by default after couple of sprints.

Reply:
OK